src/foundry/foundry.js/applications/formApplications/activeEffectConfig.d.ts

```diff
@@ -1,63 +1,76 @@
-   * @param options - Additional options which modify this application instance
+   * The Application responsible for configuring a single ActiveEffect document within a parent Actor or Item.
+   *
+   * @typeParam Options - the type of the options object
+   * @typeParam Data    - The data structure used to render the handlebars template.
+   * @typeParam O - the type of the ActiveEffect which should be managed by this form sheet
    */
-  constructor(object: O, options?: Partial<P>);
+  class ActiveEffectConfig<
+    Options extends DocumentSheet.Options = ActiveEffectConfig.Options,
+    Data extends object = ActiveEffectConfig.Data
+  > extends DocumentSheet<Options, Data, InstanceType<ConfiguredDocumentClass<typeof ActiveEffect>>> {
+    /**
+     * @override
+     * @defaultValue
+     * ```typescript
+     * foundry.utils.mergeObject(super.defaultOptions, {
+     *   classes: ["sheet", "active-effect-sheet"],
+     *   template: "templates/sheets/active-effect-config.html",
+     *   width: 560,
+     *   height: "auto",
+     *   tabs: [{navSelector: ".tabs", contentSelector: "form", initial: "details"}]
+     * });
+     * ```
+     */
+    static get defaultOptions(): ActiveEffectConfig.Options;
 
-  /** @override */
-  static get defaultOptions(): typeof FormApplication['defaultOptions'];
+    /** @override */
+    get title(): string;
 
-  /** @override */
-  get title(): string;
+    /**
+     * @override
+     *
+     * @remarks The implementation doesn't return a `Promise` but the return type includes it to allow extending
+     * classes to do that.
+     */
+    getData(options?: Application.RenderOptions): Data | Promise<Data>;
 
-  /** @override */
-  getData(options?: Application.RenderOptions): D | Promise<D>;
+    /** @override */
+    activateListeners(html: JQuery): void;
 
-  /** @override */
-  activateListeners(html: JQuery): void;
+    /**
+     * Provide centralized handling of mouse clicks on control buttons.
+     * Delegate responsibility out to action-specific handlers depending on the button action.
+     * @param event - The originating click event
+     */
+    protected _onEffectControl(event: JQuery.ClickEvent): this | void;
 
-  /**
-   * Provide centralized handling of mouse clicks on control buttons.
-   * Delegate responsibility out to action-specific handlers depending on the button action.
-   * @param event - The originating click event
-   */
-  protected _onEffectControl(event: JQuery.ClickEvent): this | void;
+    /**
+     * Handle adding a new change to the changes array.
+     */
+    protected _addEffectChange(): this;
 
-  /**
-   * Handle adding a new change to the changes array.
-   * @param button - The clicked action button
-   */
-  protected _addEffectChange(button: HTMLElement): void;
+    /**
+     * @override
+     * @param updateData - (default: `{}`)
+     */
+    _getSubmitData(updateData?: FormApplication.OnSubmitOptions['updateData']): object;
```

Comment:
It is what gets passed to `_updateObject`. Currently, in `DocumentSheet`, this is also just typed as `object`, but I suppose it should be the type that gets passed to the `update` method of the `Document` which is being configured...

Reply:
I'm not sure whether this will be right. If you look at `FormApplication._onSubmit` it is like this:

```typescript
const formData = this._getSubmitData();
await this._updateObject(event, formData);
```

So if the user overwrites both methods, he can use any data he wants to. But I'm not sure whether this is a realistic scenario.